src/networking/address.rs

```diff
@@ -13,3 +14,33 @@ impl std::fmt::Display for Address {
   }
 
 }
+
+#[cfg(test)]
+mod tests {
+  use super::*;
+  use bincode;
+
+  #[test]
+  fn test_serialization() {
```

Comment:
In this test, you basically test the serialization vs deserialization routines, i.e. you test the self-consistency of the serializer. That is fine and is a necessary first step. But it does not bind you to any *external* specification.

What you really want to add to the tests is a list of IPv8 example payloads **in a binary form* (like, hex in the source file or binary files on disk). These will anchor your implementation to an **external reference** and guarantee compatibility with original IPv8 payloads no matter how you modify the serializer.

Reply:
Yes you are right. we didnt get arount to doing this yet but it is what @NULLx76 already said and we will do this for sure.